### ISSUE TITLE
4.0 roadmap

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#189]
### ESM rule support #187 

Secretlint allow to load secretlint rule as ESM(ECMAScript modules).
You can write secretlint rule as ESM.

For more details, pleases see document.

- https://github.com/secretlint/secretlint/blob/master/docs/secretlint-rule.md

### Support `secretlint-disable` directive #96 #195 

[@secretlint/secretlint-rule-filter-comments](https://www.npmjs.com/package/@secretlint/secretlint-rule-filter-comments) support disable comment like `secretlint-disable`.

This rule is included in [@secretlint/secretlint-rule-preset-recommend](./packages/@secretlint/secretlint-rule-preset-recommend).

```
// secretlint-disable -- disable all rules

THIS IS SECRET A
THIS IS SECRET B
THIS IS SECRET C

// secretlint-enable -- enable again

// secretlint-disable-next-line @secretlint/secretlint-rule-secret-alphabet -- disable specific rule in next line
THIS IS SECRET D
THIS IS SECRET E // secretlint-disable-line -- disable current line
```

If you want to use this directive in shellscript, you can use `# secretlint-disable`.

```bash
# secretlint-disable-next-line
echo "THIS IS SECRET, BUT IT WILL BE IGNORED"
```

For more details, see <https://github.com/secretlint/secretlint/blob/master/docs/configuration.md>

### use `export cosnt creator` instead of `export default` #190 

Secretlint rule should use named export insteadof default export.
It is caused is thatDynamic Import in CommonJS is broken https://github.com/secretlint/secretlint/issues/190

If you have a secretlint rule, please change following.

```diff
- export default creator;
+ export { creator }
```

### Require Node.js 12 and update `engines` #193

Now, Secretlint requires Node.js 12+
It aims to support ECMAScript modules.

**Comment by azu** [comment]
Possibly, we want to includes #96

**Comment by azu** [comment]
#96 comment filter has beed implemented in #195

**Comment by azu** [comment]
Update preset rules #198

**Comment by azu** [comment]
```
npm install secretlint@next
```

or

```
npx @secretlint/quick-start@next "**/*"
```

Please try next version.

**Comment by azu** [comment]
#199

**Comment by azu** [comment]
https://github.com/secretlint/secretlint/releases/tag/v4.0.0 🎉
